### ISSUE TITLE
`eslint.codeActionsOnSave.mode` silently overrides `eslint.codeActionsOnSave.rules`

### ISSUE THREAD

**Issue by laverdet** [microsoft/vscode-eslint#1388]
Using dbaeumer.vscode-eslint v2.2.2 w/ eslint v8.4.1

.vscode/settings.json:
```json
{
	"eslint.codeActionsOnSave.mode": "problems",
	"eslint.codeActionsOnSave.rules": [],
	"eslint.useESLintClass": true,
}
```

.eslintrc.yaml:
```yaml
parserOptions:
  ecmaVersion: latest
  sourceType: module
rules:
  no-multiple-empty-lines: [ warn, { max: 1, maxEOF: 0 } ]
  prefer-const: warn
```

package.json:
```json
{
  "name": "eslint-test",
  "version": "1.0.0",
  "dependencies": {
    "eslint": "^8.4.1"
  }
}
```

When saving a file all fixes are applied even though the rules are explicitly set to an empty array. It's ok that these two options are mutually exclusive but the interaction between them is surprising without any kind of runtime warning or documentation of this invariant.

Additionally the documentation around `eslint.codeActionsOnSave.rules` is a little confusing. It's stated "This setting is only honored if either ESLint version 8 or greater is used or ESLint version 7 is used and the setting eslint.useESLintClass is set to true." which I mentally parsed as: `version >= 8 || (version == 7.x && useESLintClass)`. Consider: "This setting requires ESLint 7 or higher and that the eslint.useESLintClass option is enabled".

**Comment by dbaeumer** [comment]
Thanks for reporting this. I will improve the documentation around this, but `"eslint.codeActionsOnSave.mode": "problems"` will win over `"eslint.codeActionsOnSave.rules"` since otherwise the whole purpose of the mode setting is superfluous.

> This setting requires ESLint 7 or higher and that the eslint.useESLintClass option is enabled

This is not fully correct. If you use version >= 8 then the setting of `eslint.useESLintClass` is irrelevant since version 8 only ships with a ESLint Class API.

**Comment by laverdet** [comment]
> This is not fully correct. If you use version >= 8 then the setting of eslint.useESLintClass is irrelevant since version 8 only ships with a ESLint Class API.

This is not the behavior I'm observing. If `"eslint.useESLintClass": true` is omitted from settings.json then `"eslint.codeActionsOnSave.rules": []` has no effect. I'm attaching a short screen recording https://imgur.com/a/fhDqpvX

**Comment by dbaeumer** [comment]
You are correct (see https://github.com/microsoft/vscode-eslint/blob/7568b4ce8413e890e5f7d99b18845ccd00fd3d1a/server/src/eslintServer.ts#L392) However this is a bug and shouldn't be necessary

**Comment by dbaeumer** [comment]
Actually above comment is wrong. The whole code creates a ESLint class as a default at the end.

**Comment by dbaeumer** [comment]
Looking at the code I do think I fixed this, but not yet available in a new version. Do you have a repro I can clone that demos what you are seeing to ensure it works now.

**Comment by laverdet** [comment]
Sure, I made a repo which demonstrates the issue clearly:
https://github.com/laverdet/eslint-test

I've tested this on a clean installation of VS Code v1.63.1 with no user settings, and vscode-eslint v2.2.2. If you open the project and save the file `index.js` the content should not change. If you then open `.vscode/settings.json`, remove the line with content `	"eslint.useESLintClass": true,` and then save `index.js` once more the `let` will turn into a `const`.
